turtle_graphics/turtle.js

```diff
@@ -0,0 +1,324 @@
+///////////////////////////////////////////////////////////////////////////////////
+//
+//  [Homework] Turtle Graphics - 2
+//  Drawing Turtle Graphics Style
+//  In this homework, you will create a simple drawing program 
+//  inspired by Turtle Graphics. It will be a much simpler implementation 
+//  that will only accept right angles (90 degrees).
+//  Auther: Harry Ji 
+//  Date: Sep-12-2021  
+//
+///////////////////////////////////////////////////////////////////////////////////
+
+class Turtle {
+    constructor (x, y) {
+        this.x = Number(x) || 0;
+        this.y = Number(y) || 0;
+        if(this.x>=5){ 
+            this.maxX=this.x;
+        }else{
+            this.maxX=5;
+        }  
+        if(this.x<0){ 
+            this.minX =this.x;
+        }else{
+            this.minX=0;
+        }  
+        if(this.y>5){ 
+            this.maxY=this.y;
+        }else{
+            this.maxY=5;
+        }  
+        if(this.y<0 ){ 
+            this.minY=this.y;
+        }else{
+            this.minY=0;
+        }  
+        this.direction = 'x+';        
+        this.steps_array = [];
+        this.steps_array.push([this.x, this.y]);
+    }; 
+    print(file_name=''){ 
+        const turtleFootprint = (x, y) => {
+            for (let step of this.steps_array) {
+                if (step[0] === x && step[1] === y)
+                return true;
+            }
+            return false;
+        };
+        let return_string='';
+        
+        return_string += '-- BEGIN LOG\n';
+        for (let j = this.minY; j <= this.maxY; j++){
+            let row = '';
+            for (let i = this.minX; i <= this.maxX; i++){  
+                if (i === this.x && j === this.y){
+                    if (this.direction === 'x+') {
+                        row += ' > ';
+                    } else if (this.direction === 'y-') {                
+                        row += ' ^ ';
+                    } else if (this.direction === 'x-') {
+                        row += ' < ';
+                    } else if (this.direction === 'y+') {
+                        row += ' v ';
+                    }
+                    
+                }else{           
+                    if (turtleFootprint (i, j)) {   
+                        row += ' ■ ';
+                    } else {
+                        row += ' □ ';
+                    }
+                }
+            
+            }
+            return_string += row+'\n';
+        } 
+        return_string += '-- END LOG\n';
+        if(file_name ===''){
+            console.log(return_string);
+        }else{             
+            
+            fs.writeFile(file_name,return_string , function (err) {
+            if (err) return console.log(err);
```

Comment:
proper indentation please

Reply:
yes, the code is copying from the internet. so forgot indentation. thanks